fog/types/src/lib.rs

```diff
@@ -0,0 +1,34 @@
+#![no_std]
+#![deny(missing_docs)]
+
+//! This crate provides prost versions of some types from fog report server proto
+//! One reason that these prost versions are needed is so that mc-fog-report-validation
+//! doesn't depend on grpcio, as `mc-fog-api` does.
+
+extern crate alloc;
+
+use alloc::{string::String, vec::Vec};
+use prost::Message;
+use serde::{Deserialize, Serialize};
+
+/// A fog report from the report server
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize, Message)]
```

Comment:
Do these need to be `Serialize`/`Deserialize`? I believe the Protobuf ones we're switching away from in this PR wouldn't have been serde serializable. Assuming that's the case, what's the motivation for adding it?

Reply:
just copying the pattern I used in other parts of fog
one reason things become `Serialize/Deserialize` is if they have to enter or leave an enclave
it seems like it doesn't hurt anything to add this and might help something later

Reply:
Ok. I personally think we should avoid adding serde to prost struct unless there's a clear reason to. Why add the extra dependency just in case?

I believe the pattern is generally to create a separate struct specifically for the enclave boundary, no?

Reply:
> I believe the pattern is generally to create a separate struct specifically for the enclave boundary, no?

no, the pattern is to have as few structs as possible.

As a rule, structs should implement standard traits like `Debug, Clone`. In this case, it's a struct meant for serialization, so it should implement `Serialize` and `Deserialize`unless there's a reason it can't.

For some earlier structures in fog, I tried avoiding serde entirely. Most of them got serde added by other people later (for instance, the fog-kex-rng crate types.)